src/java.base/share/native/launcher/main.c

```diff
@@ -110,7 +110,23 @@ main(int argc, char **argv)
             }
         }
     }
-    JLI_CmdToArgs(GetCommandLine());
+
+    // Obtain the command line in UTF-16, then convert it to ANSI code page
+    // without the "best-fit" option
+    LPWSTR wcCmdline = GetCommandLineW();
+    int mbSize = WideCharToMultiByte(CP_ACP,
+        WC_NO_BEST_FIT_CHARS | WC_COMPOSITECHECK | WC_DEFAULTCHAR,
+        wcCmdline, -1, NULL, 0, NULL, NULL);
+    LPSTR mbCmdline = JLI_MemAlloc(mbSize);
+    if (WideCharToMultiByte(CP_ACP, WC_NO_BEST_FIT_CHARS | WC_COMPOSITECHECK | WC_DEFAULTCHAR,
```

Comment:
I think this looks okay but will need to handle GetCommandLineW, WideCharToMultiByte, or JLI_MemAlloc failing.

Reply:
IIUC, all errors should be handled with the proposed patch. On error with JLI_MemAlloc and WideCharToMultiByte, the process exits with `exit(1)`. As to `GetCommandLineW()`, there is no description of error in the MS document (https://learn.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-getcommandlinew), so I suppose no error handling on our side is needed.

Reply:
> IIUC, all errors should be handled with the proposed patch. On error with JLI_MemAlloc and WideCharToMultiByte, the process exits with `exit(1)`. 

I think I'm mostly wondering about WideCharToMultiByte as it returns 0 when it fails.

Reply:
The first call to `WideCharToMultiByte()` returns the required size for the multibyte buffer. When it returns 0, `exit(1)` is issued inside `JLI_MemAlloc()`. If the second call to WideCharToMultiByte()` returns 0, it is explicitly dealt with the patch which also issues `exit(1)`.

Reply:
> The first call to `WideCharToMultiByte()` returns the required size for the multibyte buffer. When it returns 0, `exit(1)` is issued inside `JLI_MemAlloc()`.

That's a bit subtle. I think it needs to be handled here or at least include a comment to say that it returns 0 on error which will cause JLI_MemAlloc to exit.

Reply:
OK, added the comment